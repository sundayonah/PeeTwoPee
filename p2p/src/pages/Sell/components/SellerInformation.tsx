import {
  Box,
  Circle,
  Flex,
  Img,
  Text,
  useColorModeValue,
  useMediaQuery,
} from "@chakra-ui/react";
import { StatusTag } from "../../../components/TagComponents";

import Rank from "../../../components/Rank";
import { useTranslation } from "react-i18next";

type ISeller = {
  username: string;
  img: string;
  badge: any;
  orderCompleted: number;
  orderPercentage: string;
  status: string;
};
type SellerInfo = {
  seller: ISeller;
};

export default function SellerInformation({ seller }: SellerInfo) {
  const [isMobileDevice] = useMediaQuery("(max-width: 750px)");
  const textColor = useColorModeValue("#666666", "#fff");
  const {t} = useTranslation()
  if (isMobileDevice) {
    return (
      <Flex flexDirection='column'>
        
         
          <Flex justifyContent="space-between">
            <Flex>
{seller.img ? ( 
 <Img src={seller.img} data-testid='profilePicture' />):(
 <Circle
            backgroundColor='#319EF6'
            size="50px"
            data-testid='profilePicture'
            color="#15202B"
            fontSize='16px'
            fontWeight="500"
          >
             {seller?.username[0]} 
          </Circle>
          )}
          <Text
              fontSize='14px'
              mx="18px"
              data-testid='profileUsername'
              mt="11px"
              color="#4CAFFF"

            >
              {seller.username.split(" ")[0]}{" "}
              {seller.username.split(" ")[1]?.toUpperCase()}
            </Text>
            <StatusTag status={true} mt='5' />
            </Flex>
         {seller.badge && (
              <Rank rank={seller.badge} margin="10px" />
            )}
            
          </Flex>
          

        
        <Box ml={3} mt='10px'>
          <Flex mt={1}>
            
          </Flex>
          <Text
            color={textColor}
            mt={1}
            fontSize='12px'
            data-testid='profileInformation'
          >
            {seller.orderCompleted} {" "} {t('order(s)')} | {seller.orderPercentage}%
            {" "} {t('completion')}
          </Text>
        </Box>
      </Flex>
    );
  } else {
    return (
      <Flex>
        {seller.img ? (
          <Img src={seller.img} data-testid='profilePicture' />
        ) : (
         <Circle
            backgroundColor='#319EF6'
            size="37px"
            data-testid='profilePicture'
            color="#15202B"
            fontSize='16px'
            mt="6px"
            fontWeight="500"
          >
             {seller?.username[0]} 
          </Circle>
        )}
        <Box ml={3}>
          <Flex mt={1}>
            <Text
              color='#319EF6'
              fontSize='14px'
              mr={2}
              data-testid='profileUsername'
            >
              {seller.username.split(" ")[0]}{" "}
              {seller.username.split(" ")[1]?.toUpperCase()}
            </Text>
            <StatusTag
              status={seller.status === "online" ? true : false}
              mt='2'
              mr="7px"
              
            />
            {seller.badge && (
              <Rank rank={seller.badge} />
            )}
          </Flex>
          <Text
            color={textColor}
            mt={1}
            fontSize='12px'
            data-testid='profileInformation'
          >
            {seller.orderCompleted}{" "} {t('order(s)')} | {seller.orderPercentage}%{" "}
            {t('completion')}
          </Text>
        </Box>
      </Flex>
    );
  }
}
