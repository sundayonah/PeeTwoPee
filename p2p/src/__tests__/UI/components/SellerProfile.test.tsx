import { render, screen } from '@testing-library/react';
import SellerInformation from "../../../pages/Sell/components/SellerInformation"


describe('<SellerInformation />',()=>{
  const sellerInformationNoImage ={
    username:"Gabriel Zone",
    badge:"G",
    img:"",
    orderCompleted:45,
    orderPercentage:34.6,
    status:"online"
  }
  const sellerInformation = {
    username:"John Doe",
    badge:"S",
    img:"93993",
    orderCompleted:45,
    orderPercentage:34.6,
    status:"online"
  }

  it("has the seller information with no Image",async ()=>{
     render(<SellerInformation seller={sellerInformationNoImage} />)
      const profilePicture = await screen.findByTestId("profilePicture")
        expect(profilePicture).toHaveTextContent(sellerInformationNoImage.username[0])
  })
  
  it("has the seller information",async ()=>{
      render(<SellerInformation seller={sellerInformation} />)
      const profilePicture = screen.getByTestId("profilePicture")
      const profileUsername = screen.getByTestId("profileUsername")
      const profileStatus = screen.getByTestId("profileStatus")
      const profileInformation = screen.getByTestId("profileInformation")
      const profileBadge = screen.getByTestId("profileBadge")
      

        
        expect(profilePicture).toHaveAttribute("src",sellerInformation.img)
        expect(profileUsername).toHaveTextContent(sellerInformation.username)
        expect(profileStatus).toHaveStyle(`background:#22BB33`)
        expect(profileInformation).toHaveTextContent(`${sellerInformation.orderCompleted} Order(s) | ${sellerInformation.orderPercentage}% Completion`)
        expect(profileBadge).toBeInTheDocument()
  })

  it("Matches snapshot",()=>{
    const {container} = render(<SellerInformation seller={sellerInformation} />)
    expect(container).toMatchSnapshot()
})
})