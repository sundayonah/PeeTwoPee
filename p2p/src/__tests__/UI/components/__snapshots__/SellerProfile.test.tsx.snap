// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SellerInformation /> Matches snapshot 1`] = `
<div>
  <div
    class="css-k008qs"
  >
    <img
      class="chakra-image css-0"
      data-testid="profilePicture"
      src="93993"
    />
    <div
      class="css-9n3bbc"
    >
      <div
        class="css-1euf48e"
      >
        <p
          class="chakra-text css-5r24gd"
          data-testid="profileUsername"
        >
          John Doe
        </p>
        <div
          class="css-m0ceft"
          data-testid="profileStatus"
        />
        <img
          class="chakra-image css-14gt3bc"
          data-testid="profileBadge"
          src=""
        />
      </div>
      <p
        class="chakra-text css-1vzthj9"
        data-testid="profileInformation"
      >
        45
         Order(s) | 
        34.6
        % Completion
      </p>
    </div>
  </div>
</div>
`;
